kernel/tests/dv.rs

```diff
@@ -47,3 +62,273 @@ fn non_dv_table() -> Result<(), Box<dyn std::error::Error>> {
+        use delta_kernel::engine::arrow_conversion::TryFromKernel;
+
+        let path_array = StringArray::from(vec![data_file_path]);
+
+        // Create empty map for partitionValues
+        let entries_field = Arc::new(Field::new(
+            "key_value",
+            ArrowDataType::Struct(
+                vec![
+                    Arc::new(Field::new("key", ArrowDataType::Utf8, false)),
+                    Arc::new(Field::new("value", ArrowDataType::Utf8, true)),
+                ]
+                .into(),
+            ),
+            false,
+        ));
+        let empty_keys = StringArray::from(Vec::<&str>::new());
+        let empty_values = StringArray::from(Vec::<Option<&str>>::new());
+        let empty_entries = StructArray::from(vec![
+            (
+                Arc::new(Field::new("key", ArrowDataType::Utf8, false)),
+                Arc::new(empty_keys) as ArrayRef,
+            ),
+            (
+                Arc::new(Field::new("value", ArrowDataType::Utf8, true)),
+                Arc::new(empty_values) as ArrayRef,
+            ),
+        ]);
+        let offsets = OffsetBuffer::from_lengths([0]);
+        let partition_values_array = Arc::new(MapArray::new(
+            entries_field,
+            offsets,
+            empty_entries,
+            None,
+            false,
+        ));
+
+        let size_array = Int64Array::from(vec![parquet_data_len as i64]);
+        let mod_time_array = Int64Array::from(vec![1000000i64]);
+        let num_records_array = Int64Array::from(vec![10i64]);
+
+        let stats_struct = StructArray::from(vec![(
+            Arc::new(Field::new("numRecords", ArrowDataType::Int64, true)),
+            Arc::new(num_records_array) as ArrayRef,
+        )]);
+
+        let batch = RecordBatch::try_new(
+            Arc::new(TryFromKernel::try_from_kernel(add_files_schema.as_ref())?),
+            vec![
+                Arc::new(path_array) as ArrayRef,
+                partition_values_array as ArrayRef,
+                Arc::new(size_array) as ArrayRef,
+                Arc::new(mod_time_array) as ArrayRef,
+                Arc::new(stats_struct) as ArrayRef,
+            ],
+        )?;
+
+        Box::new(ArrowEngineData::new(batch))
+    };
+
+    txn.add_files(add_metadata);
+    let commit_result = txn.commit(engine.as_ref())?;
+    assert!(matches!(
+        commit_result,
+        CommitResult::CommittedTransaction(_)
+    ));
+
+    // Step 3: Verify we can read all 10 rows before deletion
+    let snapshot = Snapshot::builder_for(table_url.clone()).build(engine.as_ref())?;
+    let scan = snapshot.scan_builder().build()?;
+    let stream = scan.execute(engine.clone())?;
+    let total_rows_before = count_total_scan_rows(stream)?;
+    assert_eq!(total_rows_before, 10, "Should have 10 rows before deletion");
+
+    // Step 4: Create deletion vectors marking rows 2, 5, and 7 as deleted
+    let mut dv = KernelDeletionVector::new();
+    dv.add_deleted_row_indexes([2, 5, 7]);
```

Comment:
updates for two files:

1.  Apply this to only the first file (with updated indices).
2. Verify deleted rows
3. Delete one more row from the same file, and delete some rows from the second file.
4. verify deleted rows again.